src/meta/src/hummock/manager/mod.rs

```diff
@@ -857,243 +855,303 @@ impl HummockManager {
-            compression_algorithm,
-            target_file_size: compact_task.target_file_size,
-            table_options: table_id_to_option
-                .into_iter()
-                .filter_map(|(table_id, table_option)| {
-                    if member_table_ids.contains(&table_id) {
-                        return Some((table_id, TableOption::from(&table_option)));
-                    }
+                .member_table_ids
+                .clone();
 
-                    None
-                })
-                .collect(),
-            current_epoch_time: Epoch::now().0,
-            compaction_filter_mask: group_config.compaction_config.compaction_filter_mask,
-            target_sub_level_id: compact_task.input.target_sub_level_id,
-            task_type: compact_task.compaction_task_type as i32,
-            split_weight_by_vnode: compact_task.input.vnode_partition_count,
-            ..Default::default()
-        };
+            let mut table_id_to_option: HashMap<u32, _> = HashMap::default();
 
-        let is_trivial_reclaim = CompactStatus::is_trivial_reclaim(&compact_task);
-        let is_trivial_move = CompactStatus::is_trivial_move_task(&compact_task);
+            for table_id in &member_table_ids {
+                if let Some(opts) = all_table_id_to_option.get(table_id) {
+                    table_id_to_option.insert(*table_id, *opts);
+                }
+            }
 
-        if is_trivial_reclaim {
-            compact_task.set_task_status(TaskStatus::Success);
-            self.report_compact_task_impl(
-                task_id,
-                Some(compact_task.clone()),
-                TaskStatus::Success,
-                vec![],
-                &mut compaction_guard,
-                None,
-            )
-            .await?;
-            tracing::debug!(
-                "TrivialReclaim for compaction group {}: remove {} sstables, cost time: {:?}",
-                compaction_group_id,
-                compact_task
-                    .input_ssts
-                    .iter()
-                    .map(|level| level.table_infos.len())
-                    .sum::<usize>(),
-                start_time.elapsed()
-            );
-        } else if is_trivial_move && can_trivial_move {
-            // this task has been finished and `trivial_move_task` does not need to be schedule.
-            compact_task.set_task_status(TaskStatus::Success);
-            compact_task
-                .sorted_output_ssts
-                .clone_from(&compact_task.input_ssts[0].table_infos);
-            self.report_compact_task_impl(
-                task_id,
-                Some(compact_task.clone()),
-                TaskStatus::Success,
-                compact_task.input_ssts[0].table_infos.clone(),
-                &mut compaction_guard,
-                None,
-            )
-            .await?;
+            while let Some(compact_task) = compact_status.get_compact_task(
```

Comment:
We should limit the max_count of trivial_tasks per round to avoid being stuck in the loop for too long.

Reply:
done